### DISCUSSION TITLE
Error: No available formula with the name

### DISCUSSION THREAD

**Discussion by 93k00c3** [Homebrew/discussions#5132] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.8-100-g0a206a4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0a206a409316bf86e8e355a1b701dbdb24e2bdbe
Last commit: 2 hours ago
Core tap JSON: 15 Feb 20:05 UTC
Core cask tap JSON: 15 Feb 20:05 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.pbkxT7erFF/org.xquartz:0
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 11.7.10-x86_64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2.1
```

### Output of `brew doctor`

```shell
Warning: You are using macOS 11.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

Cannot use brew install for any package, every time outputs something like:
`Error: No available formula with the name "f7cd5eb818b65f4cbc2720611149afd2d80daf01d521551e00e1ce48056234ed--cmake".
In formula file: /Users/administrator/Library/Caches/Homebrew/downloads/f7cd5eb818b65f4cbc2720611149afd2d80daf01d521551e00e1ce48056234ed--cmake.rb
Expected to find class F7cd5eb818b65f4cbc2720611149afd2d80daf01d521551e00e1ce48056234ed-Cmake, but only found: Cmake.`

Please help, I really need to work with brew :(

**Comment by SMillerDev** [comment]
What are you doing when that happens?

**Reply by 93k00c3** [reply]
Just trying to install ffmpeg or any package whatsoever, no formula can be found even after deleting the file listed in the error.

**Reply by 93k00c3** [reply]
Even did a reinstall of the homebrew but nothing helped.

**Reply by SMillerDev** [reply]
Can you post the full output including the command?

**Comment by ZhongRuoyu** [comment, marked as answer]
Likely caused by a change reverted in https://github.com/Homebrew/brew/pull/16684. Please do `brew update` and try again.

**Reply by 93k00c3** [reply]
Yes, the problem is now for the thank of god fixed with the new update.
Thank you for assistance, @ZhongRuoyu @SMillerDev, you dropped this: 👑
